Iqra Javed | Mongo-Friends

iqra-javed opened the pull request.

Commit:
initialized project
installed dependencies
yarn start
created friends fldr
created friendController.js & friendModel.js

Commit:
installed mongoose

Commit:
imported mongoose into server.js & friendModel.js & created friend schema

Commit:
added required flag to date in friend schema

Commit:
implemented GET friends & added required flag to age

Commit:
implemented POST friend

Commit:
implemented GET friend by ID

Commit:
implemented DELETE friend by ID

enitchals commented on friends/friendModel.js:
```diff
@@ -0,0 +1,34 @@
+    age: {
+        type: Number,
+        required: true,
+        minimum: 1, 
```
Nice! In theory the min/max shouldn't have any effect, because your Controller is checking for numbers outside this range and kicking back an error if the number is too low/high. But this is still a great exercise in using validators and thinking about your mongoose schema.

iqra-javed replied:
Good point, thanks Ellen!

enitchals commented on friends/friendController.js:
```diff
@@ -0,0 +1,76 @@
+            })
+            .catch(error => {
+                // console.log(error)
+                if (error.name ==='CastError') { // Cast Error: https://stackoverflow.com/questions/14940660/whats-mongoose-error-cast-to-objectid-failed-for-value-xxx-at-path-id
```
Nice! 🎉

enitchals reviewed (commented):
Great work on this!

Commit:
implemented PUT friend by ID & fixed 404  error handling for all methods at /:id

Commit:
fixed PUT
initially would replace anything that didnt need to be updated and was therefore not sent in the req body to null
now allows you to only send item(s) that need to be updated and sends back updated object with all other values still in tact and not nullified

Commit:
cleaned up code

Commit:
PUT for contactInfo still in progress, nullifying bug still persists

Commit:
create-react-app

Commit:
added scaffolding for react app

Commit:
installed axios

Commit:
successfully displaying friends

Commit:
added imageUrl to friends schema, displaying friends from Friend component & added some styles

Commit:
fixed nested schema issue & added notes for future reference
